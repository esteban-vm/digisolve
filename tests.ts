import type { ReactElement } from 'react'
import { create, type ReactTestRendererJSON, type TestRendererOptions } from 'react-test-renderer'

const customCreate = (ui: ReactElement, options?: TestRendererOptions) => {
  return create(ui, options).toJSON() as ReactTestRendererJSON
}

export { customCreate as create }
export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'
