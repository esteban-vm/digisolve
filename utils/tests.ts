import type { ReactElement } from 'react'
import { create } from 'react-test-renderer'

const customCreate = (element: ReactElement) => create(element).toJSON()
export { customCreate as create }

export * from '@testing-library/react'
export * from 'jest-axe'
export { default as userEvent } from '@testing-library/user-event'
