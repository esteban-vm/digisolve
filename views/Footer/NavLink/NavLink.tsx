import type { ComponentWithLink } from '@/types'
import Link from 'next/link'
import { styled } from '@/styles'

type NavLinkProps = {
  text: string
}

const NavLinkComponent: ComponentWithLink<NavLinkProps> = ({ text, url = '#', ...rest }) => {
  return (
    <li {...rest}>
      <Link href={url}>{text}</Link>
    </li>
  )
}

const NavLink = styled(NavLinkComponent)`
  margin-right: 20px;

  a:hover {
    opacity: 0.8;
  }
`

export default NavLink
