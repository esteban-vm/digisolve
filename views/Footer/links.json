{
  "navLinks": [
    { "text": "About us" },
    { "text": "Blog" },
    { "text": "Help & Support" },
    { "text": "iOS App" },
    { "text": "Android App" }
  ],
  "socialLinks": [
    { "icon": ["fab", "facebook-f"], "title": "Facebook" },
    { "icon": ["fab", "twitter"], "title": "Twitter" },
    { "icon": ["fab", "google-plus-g"], "title": "Google Plus" },
    { "icon": ["fab", "instagram"], "title": "Instagram" },
    { "icon": ["fab", "youtube"], "title": "YouTube" }
  ]
}
