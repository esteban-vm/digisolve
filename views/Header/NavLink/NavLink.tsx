import type { Component, PropsWithLink } from '@/types'
import { Link } from 'react-scroll'
import { styled } from '@/styles'

export type NavLinkProps = PropsWithLink<{ text: string }>

const NavLinkComponent: Component<NavLinkProps> = ({ text, link = '', ...rest }) => {
  return (
    <li {...rest}>
      <Link to={link} href={`#${link}`} offset={-250} delay={500} smooth>
        {text}
      </Link>
    </li>
  )
}

const NavLink = styled(NavLinkComponent)`
  display: inline-block;
  margin-right: 40px;

  a {
    :link,
    :visited {
      text-decoration: none;
      text-transform: uppercase;
      color: var(--color-white);
      outline: none;
      transition: border-bottom 100ms;
    }

    :hover,
    :active {
      border-bottom: 2px solid var(--color-primary);
    }
  }
`

export default NavLink
