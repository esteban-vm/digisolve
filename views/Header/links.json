[
  {
    "text": "Home"
  },
  {
    "text": "Contact"
  },
  {
    "text": "About"
  }
]
