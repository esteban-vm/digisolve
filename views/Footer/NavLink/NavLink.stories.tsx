import type { Meta } from '@/stories'
import { Grid } from '@/components'
import { styled } from '@/styles'
import links from '../links.json'
import NavLink from './NavLink'

const Wrapper = styled.div`
  background-color: var(--color-darker);
  padding: 50px;
  font-size: 80%;
  color: var(--color-dark);
  list-style: none;

  ul {
    float: left;

    li {
      display: inline-block;

      a {
        transition: all 500ms;

        :link,
        :visited {
          color: var(--color-dark);
          text-decoration: none;
        }
      }
    }
  }
`

export default {
  component: NavLink,
  title: 'Views/Footer/Nav Links',
  decorators: [
    (Story, context) => {
      return (
        <Wrapper>
          <Grid.Row>
            <Grid.Col isHalf>
              <Story {...context} />
            </Grid.Col>
          </Grid.Row>
        </Wrapper>
      )
    },
  ],
  parameters: {
    viewport: {
      disable: true,
    },
  },
} as Meta<typeof NavLink>

export const Example = () => {
  return (
    <ul>
      {links.navLinks.map((link) => (
        <NavLink key={crypto.randomUUID()} {...link} />
      ))}
    </ul>
  )
}
