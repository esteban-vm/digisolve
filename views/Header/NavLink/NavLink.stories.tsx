import type { Meta } from '@/stories'
import { uuid } from '@/helpers'
import { styled } from '@/styles'
import links from '@/Header/links'
import NavLink from './NavLink'

const Wrapper = styled.div`
  width: 100vw;
  height: 120px;
  background-image: linear-gradient(var(--color-overlay1), var(--color-overlay2)), url('/img/header-img.jpg');
  background-size: cover;
  background-position: top;

  ul {
    float: right;
    margin-top: 70px;
  }
`

export default {
  component: NavLink,
  title: 'Views/Header/Nav Links',
  decorators: [
    (Story, context) => {
      return (
        <Wrapper>
          <Story {...context} />
        </Wrapper>
      )
    },
  ],
  parameters: {
    backgrounds: {
      disable: true,
    },
    viewport: {
      disable: true,
    },
  },
} as Meta<typeof NavLink>

export const Example = () => {
  return (
    <ul>
      {links.map((link) => (
        <NavLink key={uuid()} {...link} />
      ))}
    </ul>
  )
}
