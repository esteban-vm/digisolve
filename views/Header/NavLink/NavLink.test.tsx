import { render, cleanup, screen, create, axe } from '@/tests'
import links from '../Header.data'
import NavLink from './NavLink'

describe('ðŸ§ª NAV LINK:', () => {
  const [testLink] = links

  describe('render tests:', () => {
    let container: HTMLElement

    beforeEach(() => {
      void ({ container } = render(
        <ul>
          <NavLink {...testLink} />
        </ul>
      ))
    })

    afterEach(cleanup)

    it('should be accessible', async () => {
      const results = await axe(container)
      expect(results).toHaveNoViolations()
    })

    it('the text of the link should be rendered', () => {
      const link = screen.getByRole('link', { name: testLink.text })
      expect(link).toBeInTheDocument()
      expect(link).toBeVisible()
    })
  })

  describe('style tests:', () => {
    it('should render with correct styles', () => {
      const tree = create(<NavLink {...testLink} />)
      expect(tree).toMatchSnapshot()
    })
  })
})
