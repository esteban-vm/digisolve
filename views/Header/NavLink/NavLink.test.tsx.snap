// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ðŸ§ª NAV LINK: style tests: should render with correct styles 1`] = `
.emotion-0 {
  display: inline-block;
  margin-right: 40px;
}

.emotion-0 a:link,
.emotion-0 a:visited {
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  color: var(--color-white);
  outline: none;
  -webkit-transition: border-bottom 100ms;
  transition: border-bottom 100ms;
}

.emotion-0 a:hover,
.emotion-0 a:active {
  border-bottom: 2px solid var(--color-primary);
}

<li
  className="emotion-0"
  id="ca898ec0-e825-11ed-b24d-893619994521"
>
  <a
    href="#home"
    onClick={[Function]}
    style={{}}
  >
    Home
  </a>
</li>
`;
